src/router/routes/genericResourceRoute.ts

```diff
@@ -127,31 +128,56 @@ export default class GenericResourceRoute {
         }
 
         if (this.operations.includes('search-type')) {
+            const handleSearch = async (res: express.Response, resourceType: string, searchParamQuery: any) => {
+                const allowedResourceTypes = await this.authService.getAllowedResourceTypesForOperation({
+                    operation: 'search-type',
+                    userIdentity: res.locals.userIdentity,
+                });
+
+                const response = await this.handler.typeSearch(
+                    resourceType,
+                    searchParamQuery,
+                    allowedResourceTypes,
+                    res.locals.userIdentity,
+                );
+                const updatedSearchResponse = await this.authService.authorizeAndFilterReadResponse({
+                    operation: 'search-type',
+                    userIdentity: res.locals.userIdentity,
+                    readResponse: response,
+                });
+                return updatedSearchResponse;
+            };
             // SEARCH
             this.router.get(
                 '/',
                 RouteHelper.wrapAsync(async (req: express.Request, res: express.Response) => {
                     // Get the ResourceType looks like '/Patient'
                     const resourceType = req.baseUrl.substr(1);
                     const searchParamQuery = req.query;
+                    const updatedSearchResponse = await handleSearch(res, resourceType, searchParamQuery);
+                    res.send(updatedSearchResponse);
+                }),
+            );
+            this.router.post(
+                '/_search',
+                RouteHelper.wrapAsync(async (req: express.Request, res: express.Response) => {
+                    // Get the ResourceType looks like '/Patient'
+                    const resourceType = req.baseUrl.substr(1);
+                    const searchParamQuery = req.query;
+                    const { body } = req;
 
-                    const allowedResourceTypes = await this.authService.getAllowedResourceTypesForOperation({
-                        operation: 'search-type',
-                        userIdentity: res.locals.userIdentity,
-                    });
+                    if (!isEmpty(body)) {
+                        mergeWith(searchParamQuery, body, (valueOne, valueTwo) => {
+                            const result: string | any[] = [];
+                            if (!isEmpty(valueOne) && !isEmpty(valueTwo) && !(valueOne === valueTwo)) {
+                                return [...new Set(result.concat(valueOne).concat(valueTwo))];
```

Comment:
concat accepts N params
```suggestion
                                return [...new Set([].concat(valueOne, valueTwo))];
```